### PR TITLE
[connectivity] Enable fetching current Wi-Fi network's BSSID

### PR DESCRIPTION
**Pull request by vargavince91**
Usage:

```dart
wifiBSSID = await _connectivity.getWifiBSSID();
```

### CODE REVIEW

**Comment by vargavince91** [comment]
I signed it!

**Comment by vargavince91** [comment]
I'm not sure why `continuous-integration/travis-ci/pr — The Travis CI build failed` is failing. I didn't see it in other merge requests.

File: packages/connectivity/lib/connectivity.dart [hunk 1]
```diff
@@ -85,6 +85,16 @@ class Connectivity {
     return wifiName;
   }
 
+  /// Obtains the wifi BSSID of the connected network
+  ///
+  /// Please note that it DOESN'T WORK on emulators (returns null).
+  ///
+  /// From android 8.0 onwards the GPS must be ON (high accuracy)
+  /// in order to be able to obtain the SSID.
```
[the remark this answers is not in the document]
Comment (vargavince91) [reply packages/connectivity/lib/connectivity.dart]: Thanks, fixed it: 8033d9e

**Comment by vargavince91** [comment]
Is there something else I should fix?

**Comment by amirh** [comment]
Thanks for contributing!

To keep the plugins repository scalable we should try to not introduce new features without e2e tests.
The plugins e2e test harness was just recently introduced, and the first e2e test for the connectivity plugin wasn't written yet, I filed https://github.com/flutter/flutter/issues/30210 to setup e2e testing for connectivity, and I think we should wait for that(or alternatively have this PR add the first connectivity feature that's tested e2e). See: https://github.com/flutter/plugins/tree/master/packages/cloud_firestore/example/test_driver for what the e2e tests might look like.

I'm marking this PR as waiting for test harness.

**Comment by vargavince91** [comment]
I signed it!

**Comment by vargavince91** [comment]
I can resolve the conflicts and add the driver tests this week (though not sure which day).

**Comment by collinjackson** [comment]
@cyanglaz I added tests and resolved conflicts, can I get your lgtm to merge this?

File: packages/connectivity/example/test_driver/connectivity.dart [hunk 2]
```diff
@@ -22,6 +22,8 @@ void main() {
         case ConnectivityResult.wifi:
           expect((_connectivity.getWifiName()), completes);
           expect((await _connectivity.getWifiIP()), isNotNull);
+          // Returns null on emulators so we just check that it doesn't throw
+          await _connectivity.getWifiBSSID();
```
Comment (cyanglaz) [inline packages/connectivity/example/test_driver/connectivity.dart]:
```suggestion
          expect((_connectivity. getWifiBSSID()), completes);
```

Comment (collinjackson) [reply packages/connectivity/example/test_driver/connectivity.dart]: good catch, you're right, this is more consistent with the rest of the test

**Pull request merged**